src/line_shapes/get_line_shapes.jl

```diff
@@ -0,0 +1,141 @@
+line bisector and width as a function of fractional line depth given in frac_depths
+note: frac_depths must be sorted from largest to smallest
+"""
+function get_line_shapes(line_list::A1, clt::AbstractChunkListTimeseries;
+                    frac_depths::A2 = 0.99:-0.01:0.05, v_width::Real=15e3,
+                    RV_estimate::Real=-5000.0) where { T1<:Real, A1<:AbstractArray{T1,1},
+                    T2<:Real, A2<:AbstractArray{T2,1}}
+    ll = copy(line_list)
+    n_lines = length(ll)
+    n_spectra = length(clt.chunk_list)
+    n_chunks = length(clt.chunk_list[1].data) # this assumes the number of chunks in every spectrum is the same
+    n_depths = length(frac_depths)
+    #allocate memory for line shapes
+    bisectors = zeros(n_lines,n_depths,n_spectra)
+    widths = zeros(n_lines,n_depths,n_spectra)
+    depths = zeros(n_lines,n_spectra)
+    #shift each entry in line list to RV estimate of star
+    ll .*= calc_doppler_factor(RV_estimate)
+    #find center of each chunk so that we can assign each line in line list to a chunk
+    chunk_centers = zeros(n_chunks)
+    for i in 1:n_chunks
+        chunk_centers[i] = (first(clt.chunk_list[1].data[i].λ) + last(clt.chunk_list[1].data[i].λ)) / 2.0
+    end
+    #pick which chunk to measure each line shape in
+    line_chunk_index = zeros(Int64,n_lines)
+    for i in 1:n_lines
+        line_chunk_index[i] = findmin(abs.((chunk_centers .- ll[i]) ./ chunk_centers))[2]
+    end
+    #measure the bisectors and widths
+    valid_lines = ones(Bool,n_lines)
+    for k in 1:n_spectra
+        #println("k = ",k)
+        for i in 1:n_lines
+            if valid_lines[i]
+                #find pixels within v_width of ll[i]
+                #println("i = ",i)
+                line_high = ll[i] * calc_doppler_factor(v_width)
+                line_low = ll[i] * calc_doppler_factor(-v_width)
+                index_low = searchsortedfirst(clt.chunk_list[k].data[line_chunk_index[i]].λ,line_low) - 1
+                index_high = searchsortedfirst(clt.chunk_list[k].data[line_chunk_index[i]].λ,line_high)
+                if (index_low < 1) | (index_high > length(clt.chunk_list[k].data[line_chunk_index[i]].λ))
+                    println("line number ",i," is not contained in chunk list ",k,". setting bisector and width functions to zeros for this line in all chunk lists.")
+                    valid_lines[i] = false
+                    bisectors[i,:,:] .= 0.0
+                    widths[i,:,:] .= 0.0
+                else
+                    #get λs and fluxes for these pixels
+                    local_λ = view(clt.chunk_list[k].data[line_chunk_index[i]].λ,index_low:index_high)
+                    local_flux = view(clt.chunk_list[k].data[line_chunk_index[i]].flux,index_low:index_high)
+                    depths[i,k] = 1.0 - minimum(local_flux[ Int(floor(length(local_flux) / 4)) : Int(ceil(length(local_flux) * 3 / 4)) ]) / maximum(local_flux)
```

Comment:
If this is intended for chunks around isolated lines, then should we fit a line model to get the depth instead of relying on minimum and maximum?